src/components/ux-reveal/ux-reveal.js

```diff
@@ -0,0 +1,22 @@
+Ractive.extend({
+	template: Ractive.defaults.templates['ux-reveal'],
+	isolated: true,
+
+	data: {
+		modalVisible: false
+	},
+
+	oninit: function () {
+
+		this.on('toggleModal', function (e) {
+			var visible = this.get('modalVisible');
+			this.set('modalVisible', !visible);
+
+			this.fire('toggleReveal', {
+				visible: this.get('modalVisible')
```

Comment:
shouldn't you just use the visible variable use before rather than get it again?